### DISCUSSION TITLE
Duckdb Driver

### DISCUSSION THREAD

**Discussion by jdgriffith** [sequelize/sequelize#17187] [Help & Questions] [answered]
Our team would find it highly beneficial if there was a Duckdb driver. This would allow us to switch to Duckdb for very fast OLAP queries when PG is not fast enough for large datasets. I'm curious how many others would find this driver useful. It would be a perfect complement to the SQLite driver for localized or in-memory databases.

**Comment by ephys** [comment, marked as answer]
That would be really cool, and we're open to feature requests. 

I must warn however that adding support for more databases requires a massive amount of initial work, as well as ongoing maintenance. For this reason, we must be mindful of how many databases we support natively.

We are, however, slowly working towards [extracting dialects outside of the core library](https://github.com/sequelize/sequelize/issues/13722) and designing a public API to allow anyone to create new dialects without needing to fork sequelize
